src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx

```diff
@@ -59,18 +63,39 @@ export function OpenSavedQueryFlyout({
   const [languageFilterOptions, setLanguageFilterOptions] = useState<string[]>([]);
   const [selectedQuery, setSelectedQuery] = useState<SavedQuery | undefined>(undefined);
   const [searchQuery, setSearchQuery] = useState(EuiSearchBar.Query.MATCH_ALL);
+  const [isLoading, setIsLoading] = useState(false);
+  const currentTabIdRef = useRef(selectedTabId);
 
   const fetchAllSavedQueriesForSelectedTab = useCallback(async () => {
-    const allQueries = await savedQueryService.getAllSavedQueries();
-    const templateQueriesPresent = allQueries.some((q) => q.attributes.isTemplate);
-    const queriesForSelectedTab = allQueries.filter(
-      (q) =>
-        (selectedTabId === 'mutable-saved-queries' && !q.attributes.isTemplate) ||
-        (selectedTabId === 'template-saved-queries' && q.attributes.isTemplate)
-    );
-    setSavedQueries(queriesForSelectedTab);
-    setHasTemplateQueries(templateQueriesPresent);
-  }, [savedQueryService, selectedTabId, setSavedQueries]);
+    setIsLoading(true);
+    const query = queryStringManager.getQuery();
+    let templateQueries: any[] = [];
+
+    // fetch sample query based on dataset type
+    if (query?.dataset?.type) {
+      templateQueries =
+        (await queryStringManager
+          .getDatasetService()
+          ?.getType(query.dataset.type)
+          ?.getSampleQueries?.()) || [];
+
+      // Check if any sample query has isTemplate set to true
+      const hasTemplates = templateQueries.some((q) => q?.attributes?.isTemplate);
+      setHasTemplateQueries(hasTemplates);
+    }
+
+    // Set queries based on the current tab
+    if (currentTabIdRef.current === 'mutable-saved-queries') {
```

Comment:
apologies i believe i missed the original design PR for template queries. 

Key Issues with Current Approach:

- Saved Query Purpose Change
  - Historically, saved queries stored exactly what users typed
  - Used for cross-application consistency
  - one major use case: Create query in Discover, apply same query in Dashboards against many saved searches and visualizations
- Cross-Plugin Impact
  - Potential unexpected behavior in Dashboards when using templated queries
  - Need to verify behavior after dataset type selection
  - Other plugins may not be aware of template functionality
Suggested Alternative Structure
- Make saved queries and saved query templates separate entities
- Create new saved object type: "saved query template"
- Keep existing saved query behavior intact
- Share UI components between both types

This maintains backwards compatibility while cleanly separating template functionality from core saved query behavior.

that said have we verified existing saved query impacts? do we need to add a saved object migration for saved queries? could we verify the impact to other plugins for example the dashboards plugin?